Assignment4/Alvaro8gb-23A257/Task07.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# **Task 07: Querying RDF(s)**
+
+# In[1]:
+
+
+get_ipython().system('pip install rdflib')
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+
+# First let's read the RDF file
+
+# In[2]:
+
+
+from rdflib import Graph, Namespace
+from rdflib.namespace import RDF, RDFS
+
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage + "/rdf/example6.rdf", format="xml")
+
+
+# In[3]:
+
+
+NS = Namespace("http://somewhere#")
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[4]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFlib:")
+for s, p, o in g.triples((None, RDFS.subClassOf, NS.LivingThing)):
+    print(s)
+
+print("SPARQL:")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE {
+    ?Subject rdfs:subClassOf ns:LivingThing.
+  }
+  ''',
+                  initNs={"rdfs": RDFS, "ns": NS}
+                  )
+for r in g.query(q1):
+    print(r.Subject)
+
+
+# **TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+# 
+
+# In[5]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFlib:")
+for s, p, o in g.triples((None, RDF.type, NS.Person)):
+    print(s)
+
+for s1, p1, o1 in g.triples((None, RDFS.subClassOf, NS.Person)):
+    for s2, p2, o2 in g.triples((None, RDF.type, s1)):
+        print(s2)
+
+q2 = prepareQuery('''
+  SELECT ?Individual WHERE {
+    ?Subject rdfs:subClassOf* ns:Person.
+    ?Individual rdf:type ?Subject
+  }
+  ''',
+                  initNs={"rdfs": RDFS, "ns": NS, "rdf": RDF}
+                  )
+
+print("SPARQL:")
+for r in g.query(q2):
+    print(r.Individual)
+
+
+# **TASK 7.3: List all individuals of "Person" or "Animal" and all their properties including their class with RDFLib and SPARQL. You do not need to list the individuals of the subclasses of person**
+# 
+
+# In[6]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFLIB")
+for c in [NS.Person, NS.Animal]:
+    for s, p, o in g.triples((None, RDF.type, NS.Person)):
```

Comment:
only type person? and animal (above)?

Reply:
thanks for the comment I will do a pull request with a better solution